extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/OpenTelemetryClientFilter.java

```diff
@@ -77,9 +77,16 @@ public void filter(final ClientRequestContext request) {
         if (parentContext == null) {
             parentContext = io.opentelemetry.context.Context.current();
         }
+
+        // For each request, we need a new OTel Context from the **current one**
+        // the parent context needs to be the one from which the call originates.
+
         if (instrumenter.shouldStart(parentContext, request)) {
             Context spanContext = instrumenter.start(parentContext, request);
-            Scope scope = QuarkusContextStorage.INSTANCE.attach(vertxContext, spanContext);
+            // Create a new scope with an empty termination callback.
+            Scope scope = (() -> {
+
+            });
```

Comment:
Given it's runtime code, I would avoid the lambda.

Reply:
I agree! Especially it's on the hot path. I wonder if I can use a constant (mutiny has a noop runnable like this).